mysql/resource_user.go

```diff
@@ -100,13 +114,34 @@ func CreateUser(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 	if v, ok := d.GetOk("auth_string_hashed"); ok {
 		hashed := v.(string)
 		if hashed != "" {
+			if authStm == "" {
+				return diag.Errorf("auth_string_hashed is not supported for auth plugin %s", auth)
+			}
 			authStm = fmt.Sprintf("%s AS '%s'", authStm, hashed)
 		}
 	}
 
-	stmtSQL := fmt.Sprintf("CREATE USER '%s'@'%s'",
-		d.Get("user").(string),
-		d.Get("host").(string))
+	var stmtSQL string
+
+	if createObj == "AADUSER" {
+		if _, uuidErr := uuid.Parse(d.Get("aad_identity").(string)); uuidErr == nil {
+			// CREATE AADUSER "mysqlProtocolLoginName"@"mysqlHostRestriction" IDENTIFIED BY "identityId"
```

Comment:
Please use single quotes in comments, just for consistency.

Reply:
Will send in minute